crates/bevy_pbr/src/meshlet/material_draw_prepare.rs

```diff
@@ -0,0 +1,404 @@
+                    format: TextureFormat::Depth16Unorm,
+                    depth_write_enabled: false,
+                    depth_compare: CompareFunction::Equal,
+                    stencil: StencilState::default(),
+                    bias: DepthBiasState::default(),
+                }),
+                multisample: MultisampleState::default(),
+                fragment: Some(FragmentState {
+                    shader: match M::meshlet_mesh_fragment_shader() {
+                        ShaderRef::Default => MESHLET_MESH_MATERIAL_SHADER_HANDLE,
+                        ShaderRef::Handle(handle) => handle,
+                        ShaderRef::Path(path) => asset_server.load(path),
+                    },
+                    shader_defs,
+                    entry_point: material_fragment.entry_point,
+                    targets: material_fragment.targets,
+                }),
+            };
+
+            let material_id = gpu_scene.get_material_id(material_id.untyped());
+
+            let pipeline_id = *cache.entry(view_key).or_insert_with(|| {
+                pipeline_cache.queue_render_pipeline(pipeline_descriptor.clone())
+            });
+            materials.push((material_id, pipeline_id, material.bind_group.clone()));
+        }
+    }
+}
+
+#[derive(Component, Deref, DerefMut, Default)]
+pub struct MeshletViewMaterialsPrepass(pub Vec<(u32, CachedRenderPipelineId, BindGroup)>);
+
+#[derive(Component, Deref, DerefMut, Default)]
+pub struct MeshletViewMaterialsDeferredGBufferPrepass(
+    pub Vec<(u32, CachedRenderPipelineId, BindGroup)>,
+);
+
+#[allow(clippy::too_many_arguments)]
+pub fn prepare_material_meshlet_meshes_prepass<M: Material>(
+    mut gpu_scene: ResMut<MeshletGpuScene>,
+    mut cache: Local<HashMap<MeshPipelineKey, CachedRenderPipelineId>>,
+    pipeline_cache: Res<PipelineCache>,
+    prepass_pipeline: Res<PrepassPipeline<M>>,
+    render_materials: Res<RenderMaterials<M>>,
+    render_material_instances: Res<RenderMaterialInstances<M>>,
+    asset_server: Res<AssetServer>,
+    mut views: Query<
+        (
+            &mut MeshletViewMaterialsPrepass,
+            &mut MeshletViewMaterialsDeferredGBufferPrepass,
+            &ExtractedView,
+            (
+                Has<NormalPrepass>,
+                Has<DepthPrepass>,
+                Has<MotionVectorPrepass>,
+                Has<DeferredPrepass>,
+            ),
+        ),
+        With<Camera3d>,
+    >,
+) where
+    M::Data: PartialEq + Eq + Hash + Clone,
+{
+    let fake_vertex_buffer_layout = &fake_vertex_buffer_layout();
+
+    for (
+        mut materials,
+        mut deferred_materials,
+        view,
+        (normal_prepass, depth_prepass, motion_vector_prepass, deferred_prepass),
+    ) in &mut views
+    {
+        if let (false, true, false, false) = (
+            normal_prepass,
+            depth_prepass,
+            motion_vector_prepass,
+            deferred_prepass,
+        ) {
```

Comment:
feel free to ignore this suggestion, just pointing out `if let` is not necessary here
```suggestion
        if (
            normal_prepass,
            depth_prepass,
            motion_vector_prepass,
            deferred_prepass,
        ) == (false, true, false, false)
        {
```